server.go

```diff
@@ -243,6 +245,36 @@ func (s *Server) register(sd *ServiceDesc, ss interface{}) {
 	s.m[sd.ServiceName] = srv
 }
 
+// ServiceMetadata returns the metadata for a service or method.
+// service should be the full service name with package, in the form of <package>.<service>.
+// method should be the method name only.
+// If only service is important, method should be an empty string.
+func (s *Server) ServiceMetadata(service, method string) interface{} {
+	// Check if service is registered.
+	if srv, ok := s.m[service]; ok {
```

Comment:
srv, ok := s.m[service]
if !ok {
  return nil
}

Go tends to prefer the “error out if you possibly can early” in order to keep the number of scope levels as small as reasonably possible.